packages/core/test/acceptance/control_flow_exploration_spec.ts

```
@@ -243,4 +243,232 @@ describe('control flow', () => {
+    it('should create, remove and move views corresponding to items in a collection', () => {
+      function App_ng_template_0_Template(rf: number, ctx: any) {
+        if (rf & 1) {
+          ɵɵtext(0);
+        }
+        if (rf & 2) {
+          const item = ctx.$implicit;
+          const idx = ctx.$index;
+          ɵɵtextInterpolate2('', item, '(', idx, ')|');
+        }
+      }
+
+      class TestComponent {
+        items = [1, 2, 3];
+
+        static ɵcmp = ɵɵdefineComponent({
+          type: TestComponent,
+          selectors: [['some-cmp']],
+          decls: 2,
+          vars: 0,
+
+          // {#for (item of items); track item; let idx = index}{{item}}({{idx}}){/for}
+          template:
+              function TestComponent_Template(rf: number, ctx: TestComponent) {
+                if (rf & 1) {
+                  ɵɵrepeaterCreate(0, App_ng_template_0_Template, 1, 2, ɵɵrepeaterTrackByIdentity);
+                }
+
+                if (rf & 2) {
+                  ɵɵrepeater(1, ctx.items);
+                }
+              },
+          encapsulation: 2
+        });
+        static ɵfac = function TestComponent_Factory(t: any) {
+          return new (t || TestComponent)();
+        };
+      }
+
+      const fixture = TestBed.createComponent(TestComponent);
+      fixture.detectChanges();
+      expect(fixture.nativeElement.textContent).toBe('1(0)|2(1)|3(2)|');
+
+      fixture.componentInstance.items.pop();
+      fixture.detectChanges();
+      expect(fixture.nativeElement.textContent).toBe('1(0)|2(1)|');
+
+      fixture.componentInstance.items.push(3);
+      fixture.detectChanges();
+      expect(fixture.nativeElement.textContent).toBe('1(0)|2(1)|3(2)|');
+
+      fixture.componentInstance.items[0] = 3;
+      fixture.componentInstance.items[2] = 1;
+      fixture.detectChanges();
+      expect(fixture.nativeElement.textContent).toBe('3(0)|2(1)|1(2)|');
+    });
+
+    it('should work correctly with trackBy index', () => {
```

Quick question about this test and the following one: if I understand correctly (based on assertions), the results of the tests would not change if `trackBy` would not be present. I'm wondering if we can update assertions to fail if `trackBy` is not present?

---

You are technically right. I will be adding much more tests after compiler is ready but modified this one to fail when tracking by sth else than expression linked to index.